### DISCUSSION TITLE
Feature request: Markdown improvements

### DISCUSSION THREAD

**Discussion by tzontron** [gsantner/markor#1664] [Ideas]
Hi, these are some additions to the markdown rendering that would be nice:

1. **syntax highlighting:** for code fences when a language is specified
2. **markdown todo checklists:** being able to tick and untick the task checkmark (same as on vscode)
3. **CSS themes included:** appreciate the ability to add custom markdown CSS but if you make changes to it when using several devices can be troublesome. The default CSS lacks some basic formatting such tables and inline code blocks (although the font is changed to monospace) which means you need to add them manually. It would be easier if the app in addition to the custom CSS has several predefined complete markdown themes you can choose from and preferably be able to switch between them, for example from a list.
4. **inject CSS from a file:** being able to add custom CSS by selecting local CSS theme file

One last request, not markdown related:

5. **hide windows:** being able to show hide windows (Notebook / ToDo / Quicknote)

Big thanks to everyone working on this wonderful and probably the best notetaking project for mobile. On a side note - please take some inspiration form [simpletask](https://github.com/mpcjanssen/simpletask-android), they still manage todo.txt handling better.

**Comment by gsantner** [comment]
1. Won't be implemented due high memory and proccessor requirements, which lowend devices can't deliver. We already discussed that multiple times on github and matrix.
2. Done via #429 , available in next update
3. Introducing themes means mostly incompatibility with other markdown parsers/renderes, this means you will see things very different on e.g. desktop vs on markor when working with a file. I'm trying to keep this difference as low as possible. If you still want such a design on all files, theres inject css ;).
4. Please make separate issue
5. More information needed, tell what you want and how it should work. Your sentence doesn't give much insight into that.

About simpletask - it's an app designed specifically for tasks, with an GUI frontend for textfiles. Markor on the other hand is a advanced texteditor, given easy tools to edit also todo.txt files. You can't really compare both as both have different goals and built around different principles.
